plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerLaunchConfigurationDelegate.java

```diff
@@ -127,6 +109,46 @@ public void serverChanged(ServerEvent event) {
     }
   }
 
+  /**
+   * Listen for the server to enter STARTED and open a web browser on the server's main page
+   */
+  protected void openBrowserPage(final ILaunchConfiguration configuration, final IServer server) {
+    if (!shouldOpenStartPage()) {
+      return;
+    }
+    final String pageLocation = determinePageLocation(server, configuration);
+    if (pageLocation != null) {
+      return;
+    }
+    final IWorkbench workbench = PlatformUI.getWorkbench();
+
+    Job openJob = new UIJob(workbench.getDisplay(), "Launching start page") {
+
+      @Override
+      public IStatus runInUIThread(IProgressMonitor monitor) {
+        if (server.getServerState() != IServer.STATE_STARTED) {
+          return Status.CANCEL_STATUS;
+        }
+        try {
+          URL url = new URL(pageLocation);
+          IWorkbenchBrowserSupport browserSupport = workbench.getBrowserSupport();
+          int style = IWorkbenchBrowserSupport.LOCATION_BAR
+              | IWorkbenchBrowserSupport.NAVIGATION_BAR | IWorkbenchBrowserSupport.STATUS;
+          browserSupport.createBrowser(style, server.getId(), server.getName(), server.getName())
+              .openURL(url);
+        } catch (PartInitException ex) {
+          // Unable to use the normal browser support, so punt to the OS
+          logger.log(Level.WARNING, "Cannot launch a browser", ex);
+          Program.launch(pageLocation);
+        } catch (MalformedURLException ex) {
+          logger.log(Level.WARNING, "Unable to determine dev_appserver URL", ex);
```

Comment:
Shouldn't this be `Level.Error`? Also the message could be `Invalid dev_appserver URL`